### PR TITLE
Coding style part4

### PR DESCRIPTION
**Pull request by kintel**
* Don't indent extern and namespaces
* Remove superfluous semicolons
* Added mod_add_long_ifdef_endif_comment
* Colon spacing rules
* Added catch brace newline rule to match else rule
* Minor tweaks to clang-format

### CODE REVIEW

**Comment by kintel** [comment]
@justbuchanan I think this should cover most bases. The next steps after this would be to add space indentation and perhaps line width limitations, and consider clang-style vs. uncrustify

File: .uncrustify.cfg [hunk 1]
```diff
@@ -77,3 +77,4 @@ sp_brace_else=force
 align_right_cmt_span=0
 
 mod_remove_extra_semicolon=true
+mod_add_long_ifdef_endif_comment=10
```
[the remark this answers is not in the document]
Comment (kintel) [reply .uncrustify.cfg]: clang-format is indeed a bit lightweight. The upside is that clang-format parses all of the C++ into an AST and hence really understand the language, while uncrustify is quite a mess with a lot of weird behavior.

**Comment by justbuchanan** [comment]
Cool, I think this is looking really good overall! Personally I find the line limits nice so that I can have two files open side-by-side on a laptop. And spaces are nice for consistency in different viewers/editors :).

I've mostly used clang-format for my projects, so I can't really compare it vs uncrustify. I've had good luck it so far, but I can't say anything bad about uncrustify - it seems like it's done well with the pull requests you've put together and it's certainly more configurable than clang-format.

**Comment by kintel** [comment]
OK, I'll merge to move forward.
I'd still like to test out clang-format a bit more. It has worked well for me in other projects (after struggling a bit to adopt a style they like) and I think that toolchain will be strong going forward + it naturally opens for adding clang-tidy down the road.

**Pull request merged**